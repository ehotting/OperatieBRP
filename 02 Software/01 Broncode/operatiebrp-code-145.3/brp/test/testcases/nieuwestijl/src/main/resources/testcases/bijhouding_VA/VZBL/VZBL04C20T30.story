Meta:
@status                 Klaar
@usecase                UCS-BY.0.VA

Narrative: Registratie verhuizing buitenland

Scenario:   voer een correcte verhuizing naar het buitenland uit. Nevenactie specifieke verstrekkingsbeperking Omschrijving derde en Gemeente verordering gevuld
            LT: VZBL04C20T30

Given alle personen zijn verwijderd
Given enkel initiele vulling uit bestand /LO3PL-VA/LO3PL-VZBL/VZBL04C20T30.xls

When voer een bijhouding uit VZBL04C20T30.xml namens partij 'Gemeente BRP 1'

Then heeft het antwoordbericht verwerking Geslaagd

Then is het antwoordbericht gelijk aan /testcases/bijhouding_VA/VZBL/expected/VZBL04C20T30.xml voor expressie //brp:bhg_vbaRegistreerVerhuizing_R

Then in kern heeft SELECT COUNT(id) FROM kern.admhnd de volgende gegevens:
| veld                      | waarde |
| count                     | 2      |

!-- Controleer betrokken personen zijn gemarkeerd als bijgehouden
Then in kern heeft select count(1)
                   from kern.his_persafgeleidadministrati hpaf
                   join kern.actie ainh on ainh.id = hpaf.actieinh
                   left join kern.actie av on av.id = hpaf.actieverval
                   left join kern.srtactie sainh on ainh.srt = sainh.id
                   left join kern.srtactie saav on av.srt = saav.id
                   left join kern.pers p on hpaf.pers = p.id
                   left join kern.admhnd a on hpaf.admhnd = a.id
                   left join kern.srtadmhnd sa on a.srt = sa.id
                   where p.bsn ='244176681' de volgende gegevens:
| veld                      | waarde |
| count                     | 2      |

!-- Controleer his_persbijhouding
Then in kern heeft select count(1)
                   from kern.his_persbijhouding hpaf
                   join kern.actie ainh on ainh.id = hpaf.actieinh
                   left join kern.actie av on av.id = hpaf.actieverval
                   left join kern.srtactie sainh on ainh.srt = sainh.id
                   left join kern.srtactie saav on av.srt = saav.id
                   left join kern.pers p on hpaf.pers = p.id
                   where sainh.naam <> 'Registratie verstrekkingsbeperking' de volgende gegevens:
| veld                      | waarde |
| count                     | 3      |

!-- Controleer adresregels in his_persadres
Then in kern heeft select sainh.naam as actieInhoud, saav.naam as actieVerval, hpad.dataanvgel, hpad.dateindegel, saaa.naam as actieAanpas, hpad.srt, hpad.rdnwijz,
                   hpad.aangadresh, hpad.dataanvadresh, hpad.identcodeadresseerbaarobject, hpad.identcodenraand, hpad.gem, hpad.nor, hpad.afgekortenor, hpad.gemdeel, hpad.huisnr,
                   hpad.huisletter, hpad.huisnrtoevoeging, hpad.postcode, hpad.wplnaam, hpad.loctenopzichtevanadres, hpad.locoms, hpad.bladresregel1,
                   hpad.bladresregel2, hpad.bladresregel3, hpad.bladresregel4, hpad.bladresregel5, hpad.bladresregel6,
                   hpad.landgebied, hpad.indpersaangetroffenopadres
                   from kern.his_persadres hpad
                   left join kern.persadres pad on pad.id = hpad.persadres
                   left join kern.pers p on p.id = pad.pers
                   join kern.actie ainh on ainh.id = hpad.actieinh
                   left join kern.actie av on av.id = hpad.actieverval
                   left join kern.actie aa on aa.id = hpad.actieaanpgel
                   left join kern.srtactie sainh on ainh.srt = sainh.id
                   left join kern.srtactie saav on av.srt = saav.id
                   left join kern.srtactie saaa on aa.srt = saaa.id
                   where p.bsn ='244176681'
                   order by hpad.dataanvgel, hpad.dateindegel de volgende gegevens:
| veld                            | waarde                                        |
| actieinhoud                     | Conversie GBA                                 |
| actieverval                     | NULL                                          |
| dataanvgel                      | 20110316                                      |
| dateindegel                     | 20170101                                      |
| actieAanpas                     | Registratie migratie                          |
| srt                             | 1                                             |
| rdnwijz                         | 4                                             |
| aangadresh                      | NULL                                          |
| dataanvadresh                   | 19980622                                      |
| identcodeadresseerbaarobject    | 0626010010016001                              |
| identcodenraand                 | 0626200010016001                              |
| gem                             | 7012                                          |
| nor                             | Baron Schimmelpenninck van der Oyelaan        |
| afgekortenor                    | S vd Oyeln                                    |
| gemdeel                         | NULL                                          |
| huisnr                          | 16                                            |
| huisletter                      | NULL                                          |
| huisnrtoevoeging                | NULL                                          |
| postcode                        | 2252EB                                        |
| wplnaam                         | Voorschoten                                   |
| loctenopzichtevanadres          | NULL                                          |
| locoms                          | NULL                                          |
| bladresregel1                   | NULL                                          |
| bladresregel2                   | NULL                                          |
| bladresregel3                   | NULL                                          |
| bladresregel4                   | NULL                                          |
| bladresregel5                   | NULL                                          |
| bladresregel6                   | NULL                                          |
| landgebied                      | 229                                           |
| indpersaangetroffenopadres      | NULL                                          |
----
| actieinhoud                     | Conversie GBA                                 |
| actieverval                     | Registratie migratie                          |
| dataanvgel                      | 20110316                                      |
| dateindegel                     | NULL                                          |
| actieAanpas                     | NULL                                          |
| srt                             | 1                                             |
| rdnwijz                         | 4                                             |
| aangadresh                      | NULL                                          |
| dataanvadresh                   | 19980622                                      |
| identcodeadresseerbaarobject    | 0626010010016001                              |
| identcodenraand                 | 0626200010016001                              |
| gem                             | 7012                                          |
| nor                             | Baron Schimmelpenninck van der Oyelaan        |
| afgekortenor                    | S vd Oyeln                                    |
| gemdeel                         | NULL                                          |
| huisnr                          | 16                                            |
| huisletter                      | NULL                                          |
| huisnrtoevoeging                | NULL                                          |
| postcode                        | 2252EB                                        |
| wplnaam                         | Voorschoten                                   |
| loctenopzichtevanadres          | NULL                                          |
| locoms                          | NULL                                          |
| bladresregel1                   | NULL                                          |
| bladresregel2                   | NULL                                          |
| bladresregel3                   | NULL                                          |
| bladresregel4                   | NULL                                          |
| bladresregel5                   | NULL                                          |
| bladresregel6                   | NULL                                          |
| landgebied                      | 229                                           |
| indpersaangetroffenopadres      | NULL                                          |

!-- Controleer de registratie van de specifieke verstrekkingsbeperking
Then in kern heeft select sainh.naam as actieInhoud, saav.naam as actieVerval, pv.omsderde, p.code as gemVerordeningPartijCode
                   from kern.persverstrbeperking pv
                   left join kern.his_persverstrbeperking hpv on hpv.persverstrbeperking = pv.id
                   left join kern.actie ainh on ainh.id = hpv.actieinh
                   left join kern.actie av on av.id = hpv.actieverval
                   left join kern.srtactie sainh on ainh.srt = sainh.id
                   left join kern.srtactie saav on av.srt = saav.id
                   left join kern.partij p on pv.gemverordening = p.id
                   left join kern.gem g on p.id = g.partij de volgende gegevens:
| veld                            | waarde                                        |
| actieinhoud                     | Registratie verstrekkingsbeperking            |
| actieverval                     | NULL                                          |
| omsderde                        | Carnavalsvereniging                           |
| gemverordeningpartijcode        | 507013                                        |

!-- Controleer of er geen gegevens voor een volledige verstrekkingsbeperking geregistreerd zijn
Then in kern heeft select count(1)
                   from kern.persindicatie pi
                   join kern.his_persindicatie hpi on hpi.persindicatie = pi.id
                   left join kern.actie ainh on ainh.id = hpi.actieinh
                   left join kern.actie av on av.id = hpi.actieverval
                   left join kern.srtactie sainh on ainh.srt = sainh.id
                   left join kern.srtactie saav on av.srt = saav.id
                   left join kern.pers p on pi.pers = p.id de volgende gegevens:
| veld                            | waarde                                        |
| count                           | 0                                             |

!-- Controleer verantwoordingsgegevens van de ActieBron
Then in kern heeft select count(1),ab.doc
                   from kern.actiebron ab join kern.actie ainh on ainh.id = ab.actie left join kern.srtactie sainh on ainh.srt = sainh.id
                   where sainh.naam in ('Registratie verstrekkingsbeperking') group by ab.doc de volgende gegevens:
| veld                      | waarde |
| count                     | 1      |