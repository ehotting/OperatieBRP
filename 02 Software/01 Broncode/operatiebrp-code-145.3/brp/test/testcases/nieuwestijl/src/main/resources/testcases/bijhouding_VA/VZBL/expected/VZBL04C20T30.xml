<brp:bhg_vbaRegistreerVerhuizing_R xmlns:brp="http://www.bzk.nl/brp/brp0200">
    <brp:stuurgegevens>
        <brp:zendendePartij>199901</brp:zendendePartij>
        <brp:zendendeSysteem>BRP</brp:zendendeSysteem>
        <brp:referentienummer>*</brp:referentienummer>
        <brp:crossReferentienummer>00000000-0000-0000-0042-200000000104</brp:crossReferentienummer>
        <brp:tijdstipVerzending>*</brp:tijdstipVerzending>
    </brp:stuurgegevens>
    <brp:resultaat>
        <brp:verwerking>Geslaagd</brp:verwerking>
        <brp:bijhouding>Verwerkt</brp:bijhouding>
        <brp:hoogsteMeldingsniveau>Geen</brp:hoogsteMeldingsniveau>
    </brp:resultaat>
    <brp:verhuizingNaarBuitenland brp:objecttype="AdministratieveHandeling">
        <brp:partijCode>507013</brp:partijCode>
        <brp:tijdstipRegistratie>*</brp:tijdstipRegistratie>
        <brp:bijgehoudenPersonen>
            <brp:persoon brp:objecttype="Persoon">
                <brp:identificatienummers>
                    <brp:burgerservicenummer>244176681</brp:burgerservicenummer>
                </brp:identificatienummers>
            </brp:persoon>
        </brp:bijgehoudenPersonen>
        <brp:bijhoudingsplan brp:objecttype="Bijhoudingsplan">
            <brp:bijhoudingsvoorstelPartijCode>507013</brp:bijhoudingsvoorstelPartijCode>
            <brp:bijhoudingsplanPersonen>
                <brp:bijhoudingsplanPersoon brp:objecttype="BijhoudingsplanPersoon">
                    <brp:persoon brp:objecttype="Persoon">
                        <brp:afgeleidAdministratief>
                            <brp:tijdstipLaatsteWijziging>*</brp:tijdstipLaatsteWijziging>
                        </brp:afgeleidAdministratief>
                        <brp:identificatienummers>
                            <brp:burgerservicenummer>244176681</brp:burgerservicenummer>
                        </brp:identificatienummers>
                        <brp:bijhouding>
                            <brp:partijCode>507013</brp:partijCode>
                        </brp:bijhouding>
                    </brp:persoon>
                    <brp:situatieNaam>Mededeling</brp:situatieNaam>
                </brp:bijhoudingsplanPersoon>
            </brp:bijhoudingsplanPersonen>
        </brp:bijhoudingsplan>
    </brp:verhuizingNaarBuitenland>
</brp:bhg_vbaRegistreerVerhuizing_R>
